src/odemis/odemisd/modelgen.py

```diff
@@ -657,42 +659,102 @@ def instantiate_microscope(self):
         self.microscope = self._instantiate_comp(self._microscope_name)
         return self.microscope
 
-    def _update_properties(self, name):
+    def _update_properties(self, comp_name):
         """
-        Set the VA values as defined in the "properties" section of the component
+        Set the VA values as defined in the "properties" section of the component. In case
+        of persistent properties (specified in "persistent" section of component) try
+        to restore the latest value from the settings file. If this fails,
+        use the VA definition in the "properties" section instead.
 
-        name (str): name of the component for which to set the VAs
+        comp_name (str): name of the component for which to set the VAs
         """
-        attrs = self.ast[name]
+        attrs = self.ast[comp_name]
+        comp = self._get_component_by_name(comp_name)
+
+        # Initialize persistent properties from settings file
+        props, _ = self.get_persistent(comp)
```

Comment:
I didn't use `._persistent_props` on purpose. Here's why:
`._persistent_props` contains the values that are read from the settings file
`.get_persistent` returns the values that are specified in the model file

It is possible that we specified persistent data in the model, but it is the first time that we open odemis with that file and the settings file is just being created, i.e. empty. In this case, it might still be a good idea to check the specified persistent data (not found in the settings file) for errors, so it fails now rather than in the end when closing odemis when we're trying to save a VA that doesn't exist. Does this make sense?

Reply:
After reading your next comment, I realize it might be better to be less strict with the settings file and not raise errors if something goes wrong, but to log a warning, also when saving values in the end, then my reasoning above doesn't apply and I can just use `._persistent_data`.

Reply:
Sorry, I was not very clear. I'm _not_ suggesting to remove `self.get_persistent()` here, but only to simplify the two ifs `if comp_name in self._persistent_props and ...` (which are later in the block). If you read `self._persistent_props[comp_name]` at the beginning, you don't need to read it again many times.